File: src/search/linear/linearSearch.test.ts
import { NO_MATCH } from "../common";
import linearSearch from "./linearSearch";

describe('Linear Search', () => {

    test('Simple Strings', () => {
        const myList = ['Hotel', 'Alpha', 'Kilo', 'Charlie', 'Echo',
            'Foxtrot', 'Delta', 'Juliet', 'Lima', 'Golf', 'Bravo']

        // Normal data which is in the list
        const charlie = linearSearch(myList, 'Charlie')
        expect(charlie).toBe(3)

        // Edge cases
        const alpha = linearSearch(myList, 'Alpha')
        expect(alpha).toBe(1)

        const lima = linearSearch(myList, 'Lima')
        expect(lima).toBe(8)

        // Not found
        const bob = linearSearch(myList, 'bob')
        expect(bob).toBe(NO_MATCH)
    })

    test('Simple Numbers', () => {
        const items = [15, 7, 1, 8, 43, 9, 81, 5];
        const found = linearSearch(items, 8);

        expect(found).toBe(3);
    });
})

File: src/search/linear/linearSearch.test.py
import unittest

from linearSearch import linearSearch, NO_MATCH


class TestLinearSearch(unittest.TestCase):

    def test_strings(self):
        myList = ['Hotel', 'Alpha', 'Kilo', 'Charlie', 'Echo',
                  'Foxtrot', 'Delta', 'Juliet', 'Lima', 'Golf', 'Bravo']
        # Normal data which is in the list
        charlie = linearSearch(myList, 'Charlie')
        self.assertEqual(charlie, 3)

        # Edge cases
        alpha = linearSearch(myList, 'Alpha')
        self.assertEqual(alpha, 1)

        lima = linearSearch(myList, 'Lima')
        self.assertEqual(lima, 8)

        # Not found
        bob = linearSearch(myList, 'bob')
        self.assertEqual(bob, NO_MATCH)


if __name__ == '__main__':
    unittest.main()
